.github/workflows/on.pr.yml

```diff
@@ -0,0 +1,154 @@
+name: "Pull Request"
+
+"on":
+  pull_request:
+    paths-ignore:
+      - '*.md'
+      - 'images/**'
```

Comment:
Separating CI into entrypoint jobs -- `on.pr.yml` and `on.push.yml` -- allows us to easily dispatch the same exact routines, but with customizations, via [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows).

This eliminates the duplicate `push`/`pr` build that was occurring before, and has the added benefit of pulling _all the pending jobs into one_, so you get one unified [job summary](https://github.com/sgammon/modern-java-practices/actions/runs/7436134602?pr=2).

Dispatching in this particular way also nicely nests the jobs in the UI, making them easy to browse:
<img width="343" alt="Screenshot 2024-01-06 at 9 13 57 PM" src="https://github.com/binkley/modern-java-practices/assets/171897/9c4e2689-8509-4841-947f-3e2245a29d7c">